From 3b96d926539740c52e63b2a8f9da23df5eedf360 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 11 Jan 2021 18:11:19 +0200
Subject: [PATCH] [SW-0] types: move types to a separate header

This is to avoid the neccesity to include preboot.h in device/common.h
files and thus introduce the circular inclusion between preboot.h and
device/common.h files.
Now one can include the types.h header instead of the whole preboot.h
and use the available types.

Change-Id: I734abdd02cc58a9db1c6be031e3885179fcd58b4
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/include/preboot.h b/include/preboot.h
index d7ac1ce..102eb59 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -27,6 +27,7 @@
 
 #include <stddef.h>
 #include <stdbool.h>
+#include <types.h>
 #include "auto_generated.h"
 
 #define VERSION_MAX_LEN		128
@@ -43,12 +44,6 @@
 #ifndef __aligned
 #define __aligned(x)	__attribute__((aligned(x)))
 #endif
-
-typedef unsigned char u8;
-typedef unsigned int u32;
-typedef unsigned short int u16;
-typedef signed int s32;
-typedef unsigned long long u64;
 
 enum hl_rst_src {
 	HL_COLD_RST,
diff --git a/include/types.h b/include/types.h
new file mode 100644
index 0000000..46219fa
--- /dev/null
+++ b/include/types.h
@@ -0,0 +1,16 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2021 HabanaLabs Ltd.
+ * All Rights Reserved.
+ *
+ */
+#ifndef _TYPEDEFS_H_
+#define _TYPEDEFS_H_
+
+typedef unsigned char u8;
+typedef unsigned int u32;
+typedef unsigned short int u16;
+typedef signed int s32;
+typedef unsigned long long u64;
+
+#endif /* _TYPEDEFS_H_ */
